tiled/adapters/mapping.py

```diff
@@ -313,6 +314,67 @@ def maybe_lower(s):
 MapAdapter.register_query(FullText, full_text_search)
 
 
+def eq(query, tree):
+    matches = {}
+    for key, value in tree.items():
+        # Find the queried key in the metadata.
+        term = value.metadata
+        for subkey in query.key.split("."):
+            if subkey not in term:
+                break
+            term = term[subkey]
+        else:
+            if term == query.value:
+                matches[key] = value
+    return tree.new_variation(mapping=matches)
```

Comment:
I feel that it would be a more "Pythonesque" approach if the method returns `None` when there is no match

Reply:
I think it's better to retain type stability. The code calling this expects a `tree`, an object with certain methods. If we _sometimes_ return `None` every piece of code that handles this will have to check if it's `None`, and if we forget somewhere, that will be an error. An empty `tree` is a valid thing to have, and it behaves sensibly. It has `len(tree) == 0` and so on.